pkg/virt-controller/watch/vmi.go

```diff
@@ -870,6 +873,17 @@ func prepareVMIPatch(oldVMI, newVMI *virtv1.VirtualMachineInstance) ([]byte, err
 		log.Log.V(3).Object(oldVMI).Infof("Patching VMI activePods")
 	}
 
+	// Apply this patch because the VMI is in an invalid state. If phase is Running or Scheduled
+	// the virt-launcher Pod needs to exist.
+	if (oldVMI.Status.Phase == virtv1.Running || oldVMI.Status.Phase == virtv1.Scheduled) &&
+		newVMI.Status.Phase == virtv1.Failed && pod == nil {
```

Comment:
This makes it look like we could potentially modify the `vmi.Status.Phase` within this controller but not want to actually write that change. If that is the case, then the logic elsewhere is wrong and needs to be addressed.

Reply:
@davidvossel Yes, it is self-defensiveness. I don't think that the handover is clear enough for people to not make mistakes. What root issue would be covered? This defensiveness would essentially make the "mistake" a no-op, or not?
Do you think this is making it too complex? Previously if you would make a mistake, moving the VMI to Failed after the handoff, the patch would not be generated.

Reply:
Yeah I think I agree with @davidvossel that this seems a little awkward.  Why not always add the patch if `oldVMI.Status.Phase != newVMI.Status.Phase`

Reply:
I also have to agree with @davidvossel. This is introducing a no-op for situations, where the wrong write to the VMI's status should have been caught during review.

Reply:
I agree with this idea `oldVMI.Status.Phase != newVMI.Status.Phase`. Thanks to all for sharing your concerns about it. I've changed the if statement, as @mhenriks suggested.